django/contrib/admin/options.py

```diff
@@ -473,9 +473,12 @@ def to_field_allowed(self, request, to_field):
             for inline in admin.inlines:
                 registered_models.add(inline.model)
 
-        for related_object in (opts.get_all_related_objects(include_hidden=True) +
-                               opts.get_all_related_many_to_many_objects()):
-            related_model = related_object.model
+        related_objects = (
+            f for f in opts.get_fields(include_hidden=True)
+            if (f.auto_created and not f.concrete)
```

Comment:
what about ```GenericForeignKey``` then? ;)
one-to-many CHECK!
not f.concrete CHECK!

In fact, there is very little difference from a ```ManyToOneRel``` reverse relation and a ```GenericReltion```!

- non concrete
- both have a reverse model
- same cardinality (m2o)
- by default, both non hidden,

The only thing that *really* differentiates the two, is that one is created by default (not user-defined) and the other one is. For this reason the flag ```auto_created``` is a key distinguisher.